populate file manager page with real data

salmanja opened the pull request:
## Describe your changes

Fixed file endpoint to populate table from backend. 

## Issue number

Resolves #1042 

## Please ensure all items are checked off before requesting a review:

- [x] I deployed the code locally.
- [x] I have performed a self-review of my code.
- [x] I have included the issue # in the PR.
- [x] I have labelled the PR correctly.
- [x] The issue I am working on is assigned to me.
- [x] I didn't use any hardcoded values (otherwise it will not scale, and will make it difficult to maintain consistency across the application).
- [x] I made sure font sizes, color choices etc are all referenced from the theme.
- [x] My PR is granular and targeted to one specific feature.
- [ ] I took a screenshot or a video and attached to this PR if there is a UI change.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - **Enhanced File Display:** Files now present standardized details—including file name, size, and upload timestamp—to ensure a more consistent view.
  - **Improved Reliability:** Optimized file retrieval incorporates better error handling for a smoother experience.
  - **Refined Table Interactions:** Updates to sorting and empty state behavior result in clearer, more intuitive file listings.
  - **Updated API Endpoint:** File fetching now utilizes a new API endpoint for improved performance.

These updates deliver an overall more reliable and user-friendly file management experience.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

samuel-coutinho commented on Clients/src/application/hooks/useFetchFiles.ts:
```diff
@@ -1,26 +1,25 @@
-              type: file.type || "N/A",
-              uploadDate: file.uploadDate ? new Date(file.uploadDate).toLocaleDateString() : "Invalid Date",
-              uploader: file.uploader || "N/A",
+              filename: file.filename,
+              uploaded_time: file.uploaded_time ? new Date(file.uploaded_time).toLocaleDateString() : "Invalid Date",
+              uploaded_by: file.uploaded_by || "N/A",
```
We could switch back to camel case here, as it is the standard convention for JavaScript.

samuel-coutinho commented on Clients/src/domain/File.ts:
```diff
@@ -1,12 +1,7 @@
 export interface FileData {
   id: string;
-  fileName: string;
+  filename: string;
+  uploaded_by: string;
+  uploaded_time: string;
```
The same as above applies here. Please switch back these names to camel case

samuel-coutinho commented on Clients/src/presentation/components/Table/FileTable/FileTable.tsx:
```diff
@@ -4,26 +4,23 @@ import { Stack, Box, Typography } from "@mui/material";
-import { File } from "../../../../domain/File";
-import { mockFiles } from "./data";
+import { FileData } from "../../../../domain/File";
+// import { mockFiles } from "./data";
```
We can remove this commented line.

samuel-coutinho commented on Clients/src/presentation/components/Table/FileTable/FileTable.tsx:
```diff
@@ -129,20 +124,20 @@ const FileTable: React.FC<FileTableProps> = ({ cols, files, onRowClick }) => {
-        file: file.name,
-        uploadDate: file.uploadDate,
-        uploader: file.uploader,
+        file: file.filename,
+        uploadDate: file.uploaded_time,
+        uploader: file.uploaded_by,
```
Please don't forget to fix these names as well.

samuel-coutinho commented on Clients/src/presentation/components/Table/FilesBasicTable/FileBasicTable.tsx:
```diff
@@ -120,9 +112,9 @@ const FileBasicTable: React.FC<FileBasicTableProps> = ({
-                <TableCell>{row.name}</TableCell>
-                <TableCell>{row.uploadDate}</TableCell>
-                <TableCell>{row.uploader}</TableCell>
+                <TableCell>{row.filename}</TableCell>
+                <TableCell>{row.uploaded_time}</TableCell>
+                <TableCell>{row.uploaded_by}</TableCell>
```
The same comment here, we should prefer to use camel case.

samuel-coutinho reviewed (changes requested):
Good work, Jamilah. Just a couple of comments about name conventions.
Could you please provide JS docs for the `useFetchFiles` hook?

samuel-coutinho commented on Clients/src/application/hooks/useFetchFiles.ts:
```diff
@@ -1,26 +1,32 @@
-              type: file.type || "N/A",
-              uploadDate: file.uploadDate ? new Date(file.uploadDate).toLocaleDateString() : "Invalid Date",
-              uploader: file.uploader || "N/A",
+              filename: file.filename,
```
Could you please change this to `fileName`? I know that the back end is sending `filename`, but the correct camel case for this attribute is `fileName`.

samuel-coutinho reviewed (changes requested):
Just a small change in the attribute name. Thanks.

MuhammadKhalilzadeh commented:
Thank you for the code @salmanja 
Please give me a heads up after it's ready for review and you're done with changes

salmanja commented:
<img width="1075" alt="Screenshot 2025-04-04 at 12 24 32 PM" src="https://github.com/user-attachments/assets/a86c0082-ebc7-4f28-8078-0f977d12f205" />

salmanja commented:
@MuhammadKhalilzadeh 
![image](https://github.com/user-attachments/assets/820f6ed1-6514-44c6-93f3-514281db2da3)

I'm getting the correct data but it's not displaying in the frontend. Currently debugging the fetch files hook.

MuhammadKhalilzadeh commented:
@salmanja please push your latest code to the branch so I can fetch, run, and debug.

salmanja commented:
> @salmanja please push your latest code to the branch so I can fetch, run, and debug.

Yes I did, debugging commits are the last two I made.

MuhammadKhalilzadeh commented:
> @MuhammadKhalilzadeh ![image](https://private-user-images.githubusercontent.com/71104867/430677611-820f6ed1-6514-44c6-93f3-514281db2da3.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM5NjgwMTgsIm5iZiI6MTc0Mzk2NzcxOCwicGF0aCI6Ii83MTEwNDg2Ny80MzA2Nzc2MTEtODIwZjZlZDEtNjUxNC00NGM2LTkzZjMtNTE0MjgxZGIyZGEzLmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQwNlQxOTI4MzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMTgyZDdjY2VlOGIzMTUwNzZlMGY0MTQyYWVmNDAzYzg5MmI2YTkwYmQ3NDZhZjdkYjAzZTIxZmY2N2Y1NDdiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.dbn1-8RiscGYShx0jMOmE5UxNDZKTKVL8XROOFTlhUo)
> 
> I'm getting the correct data but it's not displaying in the frontend. Currently debugging the fetch files hook.

---

I just fetched the code and ran. Here's what I have. Displaying the only file that I've already uploaded. Is that the expected behavior from you @salmanja ?

![Screenshot 2025-04-06 230614](https://github.com/user-attachments/assets/ec4c2367-7370-46b7-86ea-2811005b1df4)

salmanja commented:
> > @MuhammadKhalilzadeh ![image](https://private-user-images.githubusercontent.com/71104867/430677611-820f6ed1-6514-44c6-93f3-514281db2da3.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM5NjgwMTgsIm5iZiI6MTc0Mzk2NzcxOCwicGF0aCI6Ii83MTEwNDg2Ny80MzA2Nzc2MTEtODIwZjZlZDEtNjUxNC00NGM2LTkzZjMtNTE0MjgxZGIyZGEzLmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQwNlQxOTI4MzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMTgyZDdjY2VlOGIzMTUwNzZlMGY0MTQyYWVmNDAzYzg5MmI2YTkwYmQ3NDZhZjdkYjAzZTIxZmY2N2Y1NDdiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.dbn1-8RiscGYShx0jMOmE5UxNDZKTKVL8XROOFTlhUo)
> 
> > 
> 
> > I'm getting the correct data but it's not displaying in the frontend. Currently debugging the fetch files hook.
> 
> 
> 
> ---
> 
> 
> 
> I just fetched the code and ran. Here's what I have. Displaying the only file that I've already uploaded. Is that the expected behavior from you @salmanja ?
> 
> 
> 
> ![Screenshot 2025-04-06 230614](https://github.com/user-attachments/assets/ec4c2367-7370-46b7-86ea-2811005b1df4)
> 
> 

Yes that is the one expectation from filesResponse. So how come it's working on yours I'm still seeing invalid date.

MuhammadKhalilzadeh commented:
I just added a new file and it's being listed!

![Screenshot 2025-04-06 231200](https://github.com/user-attachments/assets/80030622-3897-4052-8a3c-6082f10ac971)

MuhammadKhalilzadeh commented:
@salmanja It works with having files inside the database

https://github.com/user-attachments/assets/25cc2105-3774-4834-97ce-4a5223f6ca9c

MuhammadKhalilzadeh commented:
@salmanja No errors without files ✅

![Screenshot 2025-04-06 232814](https://github.com/user-attachments/assets/7942b95f-e920-4103-a67c-18aafe35d2d0)

Well done 🙏🏻

MuhammadKhalilzadeh commented on Clients/src/application/hooks/useFetchFiles.ts:
I just reviewed the `useFetchFiles.ts` file, so let me provide some comprehensive analysis for this:

### Strengths:

1. **Well-documented**: The file has clear JSDoc comments explaining the purpose, parameters, and return values of the hook.

2. **Error handling**: The code includes proper try-catch blocks to handle potential errors during the API call.

3. **Loading state management**: The hook properly manages loading state, setting it to true before the fetch and false after completion.

4. **Data transformation**: The code transforms the API response into a consistent format with fallback values for missing data.

5. **Dependency array**: The useEffect has the correct dependency array with `projectID`, ensuring the fetch is triggered when the project ID changes.

6. **Conditional routing**: The code handles both cases where a project ID is provided or not, using different API endpoints accordingly.

---

Now, here are some improvements that I made to your file myself to decrease the issues that needs to be addressed.

### Improvements Made:

1. **Added Error State Management**:
   - Added a new state variable `error` to track and expose any errors that occur during the fetch operation
   - Updated the JSDoc to reflect that the hook now returns an error state
   - Components using this hook can now display appropriate error messages to users

2. **Implemented Request Cancellation**:
   - Added an AbortController to cancel pending requests when the component unmounts or when the project ID changes
   - Added a cleanup function to the useEffect that calls `abortController.abort()`
   - Added proper handling of AbortError to ignore it when requests are intentionally cancelled
   - Passed the abort signal to the API call

3. **Removed Console Logs**:
   - Removed all console.log statements that were cluttering the code
   - These should be replaced with proper logging in production

4. **Improved Error Handling**:
   - Added proper type checking for errors
   - Added a check to reset the error state at the beginning of each fetch
   - Properly typed the error in the catch block to fix the linter error

salmanja commented:
> Now, I have _One request for change_, and _one suggestion_.
> 
> 
> 
> 1. As a Request for change:
> 
> Uploader column must show user's full name `${name} ${surname}`
> 
> 
> 
> ![Screenshot 2025-04-07 001136](https://github.com/user-attachments/assets/40b03110-33d9-4c70-9e0d-621d47a945f0)
> 
> 
> 
> 2. And as the suggestion, adding one more column to have the download icon/button from one of these icons provided by **MUI Icons**:
> 
> What do you think? @samuel-coutinho and @gorkem-bwl 
> 
> 
> 
> ![Screenshot 2025-04-07 000954](https://github.com/user-attachments/assets/c7e363c0-e228-4321-899d-56b3e8ccb41c)
> 
> 
> 
> -----
> 
> 
> 
> Now, let's wait for @samuel-coutinho 's review as well

For #2 did you want me to create another PR to complete or in this current pull request?

gorkem-bwl commented:
@MuhammadKhalilzadeh we should refrain from using Material icons. In our style guide (Figma) we already have all the required icons. That also goes with the icons in the button which are lately introduced - they must be replaced with the icons we have on Figma. Any outside icon doesn't go well with what we have in terms of style.

salmanja commented:
> I just added a new file and it's being listed!
> 
> 
> 
> ![Screenshot 2025-04-06 231200](https://github.com/user-attachments/assets/80030622-3897-4052-8a3c-6082f10ac971)
> 
> 

I updated the containers and now it works.

MuhammadKhalilzadeh commented:
> > Now, I have _One request for change_, and _one suggestion_.
> > 
> > 1. As a Request for change:
> > 
> > Uploader column must show user's full name `${name} ${surname}`
> > ![Screenshot 2025-04-07 001136](https://private-user-images.githubusercontent.com/140876993/430737901-40b03110-33d9-4c70-9e0d-621d47a945f0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQwMDM3NjcsIm5iZiI6MTc0NDAwMzQ2NywicGF0aCI6Ii8xNDA4NzY5OTMvNDMwNzM3OTAxLTQwYjAzMTEwLTMzZDktNGM3MC05ZTBkLTYyMWQ0N2E5NDVmMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQwN1QwNTI0MjdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMjQxNzRjYWYwMGE3ZGFmYTM2MWYxYWVlYzkzMWNhNGM2NTg2MzQwMjVkYjE3MjgyYmMxNTY0Y2M0ZDdlMjBkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.F6rIDMsZoaEkBcuIfVkKmpO7BJLq_niX54ikoUwkIUg)
> > 
> > 2. And as the suggestion, adding one more column to have the download icon/button from one of these icons provided by **MUI Icons**:
> > 
> > What do you think? @samuel-coutinho and @gorkem-bwl
> > ![Screenshot 2025-04-07 000954](https://private-user-images.githubusercontent.com/140876993/430738076-c7e363c0-e228-4321-899d-56b3e8ccb41c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQwMDM3NjcsIm5iZiI6MTc0NDAwMzQ2NywicGF0aCI6Ii8xNDA4NzY5OTMvNDMwNzM4MDc2LWM3ZTM2M2MwLWUyMjgtNDMyMS04OTlkLTU2YjNlOGNjYjQxYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQwN1QwNTI0MjdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYzhiMmE3ZmJlMDhmMjBjOTFiZjhmN2I0NDA0MjNkMTczZDkxZjIzMWRhOWFmMTUwOTk0MTY4YjVjZGRmMGM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.DGHJB_24c0Xn7MuNbnfJU0mLb0yzsJZMFbsnyJZD2Dc)
> > 
> > Now, let's wait for @samuel-coutinho 's review as well
> 
> For #2 did you want me to create another PR to complete or in this current pull request?

Yes please.

MuhammadKhalilzadeh commented:
> @MuhammadKhalilzadeh we should refrain from using Material icons. In our style guide (Figma) we already have all the required icons. That also goes with the icons in the button component which is lately modified - they must be replaced with the icons we have on Figma. Any outside icon doesn't go well with what we have in terms of style.

Great! then we already have what we want in our designs! @salmanja only needs to export that as **SVG** from the Figma.

MuhammadKhalilzadeh commented:
Here's what we have from our Figma style guide:

![Screenshot 2025-04-07 090005](https://github.com/user-attachments/assets/3478fd05-fc2a-4da6-9874-e367a8843599)

gorkem-bwl commented:
@samuel-coutinho we are mostly in line with @salmanja here. Basically we can start small by adding the following items in the list: 

- File name
- User who uploaded the file
- Upload time
- Action (only download for now)

As time goes by, we can add more features on top, but what Jamilah has come up with as an initial skeleton is a good start. 

The download section would be a drop down, and there would be only "Download" option for now.

samuel-coutinho commented on Clients/src/domain/File.ts:
```diff
@@ -8,5 +8,3 @@ export interface FileData {
   size: number;
   data?: Blob;
 }
-
```
Please remove the `name` attribute since the correct one is `fileName`

samuel-coutinho reviewed (approved):
LGTM, thanks Jamilah. Do you have any other comments or suggestions @MuhammadKhalilzadeh ? If not, we can merge this one.
